### DISCUSSION TITLE
The output is internal representation of a Feynman amplitude

### DISCUSSION THREAD

**Discussion by SiWang4phys** [FeynCalc/feyncalc#241] [Q&A] [answered]
Hi,

I was using a model generated by FeynRules, applied in FeynArts then FeynCalc.
It generated the Feynman diagram, but the amplitude is not right. Also, I cannot use DiracSimplify.
I generated the amplitude as:
`amp[0] = FCFAConvert[CreateFeynAmp[DiagramExtract[diags, {1, 2}]], 
   IncomingMomenta -> {p1, p2}, OutgoingMomenta -> {k1, k2}, 
   UndoChiralSplittings -> True, 
   ChangeDimension -> 4,(*List\[Rule]False,*)SMP -> True, 
   Contract -> True, 
   FinalSubstitutions -> {SMP["m_e"] -> 0, SMP["m_mu"] -> 0}] `
The output is :
`FCFAConvert(FeynAmpList(Process->(F(2,{1})	p1	Me	{-Q,LeptonNumber}
-F(2,{1})	p2	Me	{Q,-LeptonNumber}
)->(F(2,{3})	k1	MTA	{-Q,LeptonNumber}
-F(2,{3})	k2	MTA	{Q,-LeptonNumber}
),Model->{SMEFT_FA/SMEFTybg_all},GenericModel->{SMEFT_FA/SMEFTybg_all},AmplitudeLevel->{Particles},ExcludeParticles->{S,S(1),S(2),-S(3),S(3)},ExcludeFieldPoints->{},LastSelections->{})\[InvisibleApplication](FeynAmp(GraphID(Topology==1,Generic==1,Particles==1,Number==1),Integral[],(\[CurlyPhi](-(p2),Me)).(I gc81 Overscript[\[Gamma], _]^Lor1.Overscript[\[Gamma], _]^6+I gc81 Overscript[\[Gamma], _]^Lor1.Overscript[\[Gamma], _]^7+1/(2 Sqrt[2] LambdaSMEFT^2) I gc81 vev`
And more.
What is the problem?

Thank you,
Si

**Comment by vsht** [comment]
Hi,

it looks like FeynCalc is not loaded at all, since FCFAConvert is not being evaluated. 

Otherwise, please provide the complete code including all files so that one can reproduce the issue.

Cheers,
Vlad

**Comment by SiWang4phys** [comment]
Hi Vlad,

Here is Load FeynCalc, I guess there may be a problem, as I created it myself:
' If[$FrontEnd === Null, $FeynCalcStartupMessages = False;
  Print[description];];
If[$Notebooks === False, $FeynCalcStartupMessages = False];
Get["FeynCalc`"]
SetOptions[FourVector, FeynCalcInternal -> False]; 

<< FeynArts`
InitializeModel["/home/si/Dropbox/Linux/HEPSoftware/mathematica/\
FeynArt_calc/FeynArts-3.11/Models/SMEFT_FA/SMEFTybg_all", 
 GenericModel -> 
  "/home/si/Dropbox/Linux/HEPSoftware/mathematica/FeynArt_calc/\
FeynArts-3.11/Models/SMEFT_FA/SMEFTybg_all"]
'

Later is:
`MakeBoxes[p1, TraditionalForm] := "\!\(\*SubscriptBox[\(p\), \(1\)]\)";
MakeBoxes[p2, TraditionalForm] := "\!\(\*SubscriptBox[\(p\), \(2\)]\)";
MakeBoxes[k1, TraditionalForm] := "\!\(\*SubscriptBox[\(k\), \(1\)]\)";
MakeBoxes[k2, TraditionalForm] := "\!\(\*SubscriptBox[\(k\), \(2\)]\)";

diags = InsertFields[
   CreateTopologies[0, 
    2 -> 2], {F[2, {1}], -F[2, {1}]} -> {F[
      2, {3}], -F[2, {3}]},
   InsertionLevel -> {Particles}, 
   Model -> FileNameJoin[{"SMEFT_FA", "SMEFTybg_all"}], 
   GenericModel -> FileNameJoin[{"SMEFT_FA", "SMEFTybg_all"}], 
   ExcludeParticles -> {S}];
Paint[DiagramExtract[diags, {1, 2}], ColumnsXRows -> {2, 1}, 
  Numbering -> Simple, SheetHeader -> None, ImageSize -> {256, 256}];
`

Cheers,
Si

**Comment by vsht** [comment]
Looks like your FeynArts is not patched, so it will not work. Please reinstall FeynCalc using the automatic installer
and select FeynArts as well. The model must be patched too, cf.

https://feyncalc.github.io/FeynCalcBookDev/Extra/CustomModels.html

**Comment by SiWang4phys** [comment]
Hi,
I have done reinstall before:
`Import["https://raw.githubusercontent.com/FeynCalc/feyncalc/master/\
install.m"]
 InstallFeynCalc[]`
And used the patched FeynCalc10.0.0 and FeynArts 3.11.
But after I add
`FAPatch[PatchModelsOnly -> True, 
 FAModelsDirectory -> 
  "/home/si/Dropbox/Linux/HEPSoftware/mathematica/FeynArt_calc/\
FeynArts-3.11/Models/SMEFT_FA"]`
It shows patched successfully.
It still outputs the same thing.

One error I thought may have an effect.
After loading the FeynCalc and FeynArts,
The error is:
"Symbol \!\(TraditionalForm\`\"Loop\"\) appears in multiple contexts \!\(TraditionalForm\`{\"FeynArts`\", \"FeynCalc`\"}\); definitions in context \!\(TraditionalForm\`\"FeynArts`\"\) may shadow or be shadowed by other definitions."

Is there a complete standard code for using a new model? Then I can refer and change accordingly.

Cheers,
Si

**Reply by vsht** [reply, marked as answer]
If you installed FC using automatic installer, then you *must* use the FeynArts version inside the FeynCalc directory,
because it is the only one that gets patched. And then its loaded via

```
$LoadAddOns = {"FeynArts"};
<< FeynCalc`
```
You cannot just load some random FeynArts version from another directory and expect things to work.

**Comment by SiWang4phys** [comment]
I think you are right.
And also made other changes.
I looked at the other person's complete code, and it outputs the right representation.
Thanks,
Si
